src/shadowbox/server/manager_service.ts

```diff
@@ -146,6 +156,41 @@ export class ShadowsocksManagerService {
     }
   }
 
+  // Sets the default ports for new access keys
+  public async setPortForNewAccessKeys(req: RequestType, res: ResponseType, next: restify.Next):
+      Promise<void> {
+    try {
+      logging.debug(`setPort[ForNewAccessKeys request ${JSON.stringify(req.params)}`);
+      if (!req.params.port) {
+        return next(
+            invalidPortArgument(`Expected a port argument but found none. Request: ${req}`));
+      }
+
+      const port = req.params.port;
+      if (Number.isNaN(port)) {
+        return next(invalidPortArgument(`Expected an numeric port, instead got ${port}`));
+      }
+      if (port < 1 || port > 65535) {
+        return next(invalidPortArgument(
+            `Expected a positive port number up to 65535, instead got ${port}`));
+      }
+
+      await this.accessKeys.setPortForNewAccessKeys(port);
+      this.serverConfig.data().portForNewAccessKeys = port;
+      this.serverConfig.write();
+      res.send(HttpSuccess.NO_CONTENT);
+      next();
+    } catch (error) {
+      logging.error(error);
+      if (error instanceof errors.InvalidPortNumber) {
+        return next(invalidPortArgument(error.message));
```

Comment:
Could you use `resitfy.BadRequestError` here instead? It makes it easier to see what HTTP error is being returned.

Reply:
That's what invalidPortArgument does, I just moved it into a function since there are a few different places dealing with invalid port numbers.

Reply:
I'd still like to be able to easily see what the HTTP error we are returning, since that's part of the API.
Changing the calls to BadRequestError would make that clearer, I don't think the duplication is a big deal.

Maybe you want a IsValidPort method that validates the port and raises InvalidPortNumber. That would remove the duplication and improve readability.